packages/did-eth-registry/test/EthereumDIDRegistryDelegate.t.sol

```diff
@@ -0,0 +1,422 @@
+    }
+
+    function testRevokeDelegateSignedBadSignature() public {
+        vm.prank(SIGNER);
+        registry.addDelegate(SIGNER, "attestor", SIGNER2, 1 days);
+        (uint8 v, bytes32 r, bytes32 s) = signRevoke(SIGNER, SIGNER2, address(registry), PRIVATE_KEY);
+        vm.expectRevert("bad_signature");
+        vm.prank(SIGNER3);
+        registry.revokeDelegateSigned(SIGNER, v, r, s, "attestor", SIGNER3);
+        vm.expectRevert("bad_signature");
+        vm.prank(SIGNER3);
+        registry.revokeDelegateSigned(SIGNER3, v, r, s, "attestor", SIGNER);
+    }
+
+    function testRevokeDelegateSignedWrongSignatureKey() public {
+        vm.prank(SIGNER);
+        registry.addDelegate(SIGNER, "attestor", SIGNER2, 1 days);
+        (uint8 v, bytes32 r, bytes32 s) = signRevoke(SIGNER, SIGNER2, address(registry), PRIVATE_KEY2);
+        vm.expectRevert("bad_signature");
+        vm.prank(SIGNER);
+        registry.revokeDelegateSigned(SIGNER, v, r, s, "attestor", SIGNER2);
+    }
+
+    function testRevokeDelegateSignedWrongData() public {
+        vm.prank(SIGNER);
+        registry.addDelegate(SIGNER, "attestor", SIGNER2, 1 days);
+        (uint8 v, bytes32 r, bytes32 s) = signRevoke(SIGNER, SIGNER, address(registry), PRIVATE_KEY);
+        vm.expectRevert("bad_signature");
+        vm.prank(SIGNER);
+        registry.revokeDelegateSigned(SIGNER, v, r, s, "attestor", SIGNER2);
+    }
+
+    function testRevokeDelegateSignedWithIncorrectNonce() public {
+        vm.prank(SIGNER);
+        registry.addDelegate(SIGNER, "attestor", SIGNER2, 1 days);
+        bytes memory message = abi.encodePacked(bytes("revokeDelegate"), bytes32("attestor"), SIGNER2);
+        address idOwner = registry.identityOwner(SIGNER2);
+        uint256 ownerNonce = registry.nonce(idOwner) + 0x1; // not expected
+        (uint8 v, bytes32 r, bytes32 s) = VmDigest.signData(
+            vm,
+            SIGNER,
+            address(registry),
+            PRIVATE_KEY,
+            message,
+            ownerNonce
+        );
+        vm.prank(SIGNER);
+        vm.expectRevert("bad_signature");
+        registry.revokeDelegateSigned(SIGNER, v, r, s, "attestor", SIGNER2);
+    }
+
+    /**
+     * @dev Sign data with private key
+     * @param _identity Identity address
+     * @param _delegate Delegate address
+     * @param _registry DID Registry address
+     * @param _privateKey Private key
+     * @param _message Message to sign
+     * @return v Signature v
+     * @return r Signature r
+     * @return s Signature s
+     */
+    function signData(
+        address _identity,
+        address _delegate,
+        address _registry,
+        bytes memory _privateKey,
+        bytes memory _message
+    )
+        internal
+        view
+        returns (
+            uint8 v,
+            bytes32 r,
+            bytes32 s
+        )
+    {
+        address idOwner = registry.identityOwner(_delegate);
+        uint256 ownerNonce = registry.nonce(idOwner);
```

Comment:
This seems wrong. The `nonce` used should be the one from the identity owner of the `_identity`, not the `_delegate`. Perhaps some tests are passing for the wrong reasons (all nonces start at 0)?

```suggestion
        address idOwner = registry.identityOwner(_identity);
        uint256 ownerNonce = registry.nonce(idOwner);
```

Reply:
Your correct.  I applied and fixed that.